src/hosted_db/sandbox.rs

```diff
@@ -0,0 +1,120 @@
+/* Retrieved and modified from
+  https://raw.githubusercontent.com/Defelo/sandkasten/83f629175d02ebc70fbb16b8b9e05663ea67ccc7/src/sandbox.rs
+  On December 6, 2023.
+  Original license: MIT.
+*/
+
+use crate::error::AybError;
+use serde::{Deserialize, Serialize};
+use std::env::current_exe;
+use std::fs::canonicalize;
+use std::{
+    path::{Path, PathBuf},
+    process::Stdio,
+};
+use tokio::io::{AsyncReadExt, BufReader};
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct RunResult {
+    /// The exit code of the processes.
+    pub status: i32,
+    /// The stdout output the process produced.
+    pub stdout: String,
+    /// The stderr output the process produced.
+    pub stderr: String,
+}
+
+pub async fn run_in_sandbox(
+    nsjail: &Path,
+    db_path: &PathBuf,
+    query: &str,
+) -> Result<RunResult, AybError> {
+    let mut cmd = tokio::process::Command::new(nsjail);
+
+    cmd.arg("--really_quiet") // log fatal messages only
```

Comment:
Interestingly, initially I tried importing Sandkasten so I could use its abstraction, but that broke down as that library wasn't designed as an abstraction for other libraries. So I copied in the parts that I needed here (thus the copyright notice), and realized that the amount of abstraction was too high for something `ayb` would have one concrete use case for. The abstracted code was more difficult to follow, since `ayb` isn't `nsjail-wrapper`, and so I went with a more concrete single use case here.

I generally avoid abstractions that have only one concrete instance since I can't vet the abstraction without multiple examples. My bias is to follow that general avoidance in this case, but if there's some specific benefit you can think of that outweighs the cost of indirection, I'm open to hearing it! :)

Reply:
You're right, that abstraction wouldn't make much sense in our case.
I think the comments are enough, I'll mark this as resolved :)

Reply:
Hooray! Thank you!